services/browsers/api_test.go

```diff
@@ -0,0 +1,130 @@
+package browsers
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"sort"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/stretchr/testify/require"
+)
+
+func setupTestDB(t *testing.T) (*Database, func()) {
+	tmpfile, err := ioutil.TempFile("", "browsers-tests-")
+	require.NoError(t, err)
+	db, err := InitializeDB(tmpfile.Name(), "browsers-tests")
+	require.NoError(t, err)
+	return db, func() {
+		require.NoError(t, db.Close())
+		require.NoError(t, os.Remove(tmpfile.Name()))
+	}
+}
+
+func setupTestAPI(t *testing.T) (*API, func()) {
+	db, cancel := setupTestDB(t)
+	return &API{s: &Service{db: db}}, cancel
+}
+
+func TestBrowsersOrderedNewestFirst(t *testing.T) {
+	api, cancel := setupTestAPI(t)
+	defer cancel()
+
+	browsers := []*Browser{
+		{
+			ID:        "1",
+			Name:      "first",
+			Dapp:      true,
+			Timestamp: hexutil.Uint64(10),
+		},
+		{
+			ID:        "2",
+			Name:      "second",
+			Dapp:      true,
+			Timestamp: hexutil.Uint64(50),
+		},
+		{
+			ID:        "3",
+			Name:      "third",
+			Dapp:      true,
+			Timestamp: hexutil.Uint64(100),
+		},
+	}
+	// insert in reverse order for a clean test
```

Comment:
This comment doesn't seem to reflect what it is doing.

Reply:
right, forgot to cleanup